src/main/java/user_use_case/gateways/UserGateway.java

```diff
@@ -204,17 +204,83 @@ public User getUser(String username) {
         return null;
     }
 
-    //@Override
-    //public void removeUser(User user) {
-        // DOES NOT WORK CURRENTLY
+    @Override
+    public void removeUser(User user) {
         // Change username of user to "Removed" in database
         // functionality will be equivalent to if user was from database
-        //String new_username = "Removed";
 
-        //user.setUsername(new_username);
+        String given_username = user.getUsername();
+
+        String new_username = "Removed";
+
+        File old_file = new File(NAME_OF_USER_DATABASE);
+        File new_file = new File(TEMP_FILE);
+
+        String username;
+        String password;
+        String past_reviews;
+        String likedReviews;
+        String received_reports;
+        String banned;
+        String owner;
+        String owned_restaurants;
+
 
-        //updateUser(user);
-    //}
+        try {
+            FileWriter fw = new FileWriter(new_file, true);
+            BufferedWriter bw = new BufferedWriter(fw);
+            File file = new File(NAME_OF_USER_DATABASE);
+            Scanner scanner = new Scanner(file);
+            String delimiter = ", ";
+            String line;
+
+            while (scanner.hasNext()) {
+                line = scanner.nextLine();
+                String[] values = line.split(delimiter);
+
+                username = values[0];
+                password = values[1];
+                past_reviews = values[2];
+                likedReviews = values[3];
+                received_reports = values[4];
+                banned = values[5];
+                owner = values[6];
+                owned_restaurants = values[7];
+
+
+                // replace the old values with the new values
+                String line1;
+                if (given_username.equals(username)) {
+                    line1 = String.join(", ", new_username, password, past_reviews,
```

Comment:
Since it is removing the User it shouldn't have new values to put in, it should just skip and go to the next value